tests/TestCase/Command/PluginUnloadCommandTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP :  Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP Project
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Shell\Task;
+
+use Cake\Console\Command;
+use Cake\TestSuite\ConsoleIntegrationTestTrait;
+use Cake\TestSuite\TestCase;
+
+/**
+ * PluginUnloadCommandTest class
+ */
+class PluginUnloadCommandTest extends TestCase
+{
+    use ConsoleIntegrationTestTrait;
+
+    /**
+     * @var string
+     */
+    protected $app;
+
+    /**
+     * @var string
+     */
+    protected $originalAppContent;
+
+    /**
+     * setUp method
+     *
+     * @return void
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->app = APP . DS . 'Application.php';
+
+        $this->originalAppContent = file_get_contents($this->app);
+
+        $this->useCommandRunner();
+        $this->setAppNamespace();
+    }
+
+    /**
+     * tearDown method
+     *
+     * @return void
+     */
+    public function tearDown(): void
+    {
+        parent::tearDown();
+        $this->clearPlugins();
+
+        file_put_contents($this->app, $this->originalAppContent);
+    }
+
+    /**
+     * testUnload
+     *
+     * @return void
+     */
+    public function testUnload()
+    {
+        $plugin1 = "\$this->addPlugin('TestPlugin', ['bootstrap' => false, 'routes' => false]);";
+        $plugin2 = "\$this->addPlugin('TestPluginTwo', ['bootstrap' => false, 'routes' => false]);";
+        $this->addPluginToApp($plugin1);
+        $this->addPluginToApp($plugin2);
+
+        $contents = file_get_contents($this->app);
+        $this->assertStringContainsString($plugin1, $contents);
+
+        $this->exec('plugin unload TestPlugin');
+
+        $this->assertExitCode(Command::CODE_SUCCESS);
+        $contents = file_get_contents($this->app);
+
+        $this->assertStringNotContainsString($plugin1, $contents);
+        $this->assertStringContainsString($plugin2, $contents);
+    }
+
+    /**
+     * Data provider for various forms.
+     *
+     * @return array
+     */
+    public function variantProvider()
+    {
+        return [
+            //  $this->addPlugin('TestPlugin', [
```

Comment:
The commented code needs to be removed.

Reply:
these are sample to show how would be value provider items 
see old code
https://github.com/cakephp/cakephp/blob/b879ff9e15c8519ecda8e8a2708f7aa08493e6af/tests/TestCase/Shell/Task/UnloadTaskTest.php#L110-L114

Reply:
Ah, never mind than.